src/risk_distributions/risk_distributions.py

```diff
@@ -352,81 +334,95 @@ class Normal(BaseDistribution):
+                'gumbel': Gumbel,
+                'invgamma': InverseGamma,
+                'invweibull': InverseWeibull,
+                'llogis': LogLogistic,
+                'lnorm': LogNormal,
+                'mgamma': MirroredGamma,
+                'mgumbel': MirroredGumbel,
+                'norm': Normal,
+                'weibull': Weibull}
+
+    @classmethod
+    def get_valid_distributions(cls, weights: pd.DataFrame, mean: DistParamValue,
+                                sd: DistParamValue) -> Tuple[np.ndarray, Dict]:
         """Produces a distribution that filters out non convergence errors and rescales weights appropriately.
         Can specify either data, params, or (mean and standard deviation), but not multiple.
 
         Parameters
         ----------
         weights :
-            A list of normalized distribution weights indexed by distribution type.
-        distribution_map :
-            Mapping between distribution name and distribution class.
-        data :
-            Table where each row has a `mean` and `standard_deviation` for a single distribution.
-        params :
-            Dictionary of pre-calculated parameters for distribution.
+            A single-row dataframe of normalized distribution weights, each column representing
+            a different distribution type.
         mean :
             Mean value for a single distribution or series of values, each for a single distribution.
-        std_dev :
+        sd :
             Standard deviation value for a single distribution or series of values, each for a single distribution.
 
         Returns
         -------
             Rescaled weights and the subset of the distribution map corresponding to convergent distributions.
         """
-        dist = dict()
-        for key in distribution_map:
+
+        distributions = dict()
+        distribution_map = cls.get_distribution_map()
+        distribution_types = list(set(distribution_map.keys()) & set(weights.columns))
+        weights = weights[distribution_types].iloc[0]
+        weights = weights/np.sum(weights)
+        for name in distribution_types:
             try:
-                dist[key] = distribution_map[key](data=data, params=params, mean=mean, std_dev=std_dev)
+                distributions[name] = distribution_map[name](mean=mean, sd=sd)
             except NonConvergenceError as e:
                 if weights[e.dist] > 0.05:
                     weights = weights.drop(e.dist)
                 else:
-                    raise NonConvergenceError(f'Divergent {key} distribution has weights: {100*weights[key]}%', key)
+                    raise NonConvergenceError(f'Divergent {key} distribution has weights: {100*weights[name]}%', name)
 
-        return weights/np.sum(weights), dist
+        return weights/np.sum(weights), distributions
 
     def pdf(self, x: pd.Series) -> Union[np.ndarray, pd.Series]:
         return np.sum([self.weights[name] * dist.pdf(x)
```

Comment:
github doesn't allow me to add comments on the parts not changed :( Is there a reason why we only check whether x is empty for ppf? or did I leave it like that 6 months ago? (probably i did, i guess..)

Reply:
Oh good point. That's how it was originally but I can definitely change it to check for x being empty for pdf as well